### PR TITLE
Add data structure

### PR DESCRIPTION
**Pull request by LumisD**
Please review

### CODE REVIEW

File: app/src/main/java/com/lumisdinos/gameoffifteen/data/GameRepositoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+class GameRepositoryImpl @Inject constructor(
+    private val daoDB: DaoDB,
+    private val gameDataMapper: GameDataMapper
+) : GameRepository {
+
+    override suspend fun getAllGames(): List<GameModel> {
+        return daoDB.getAllGames().map { with(gameDataMapper) { it.fromEntityToDomain() } }
+    }
+
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/lumisdinos/gameoffifteen/data/GameRepositoryImpl.kt:15]: This way:
override suspend fun getAllGames(): List<GameModel> =
        daoDB.getAllGames().map { with(gameDataMapper) { it.fromEntityToDomain() } }

**Review by vladislav-v-rusakov — APPROVED** [review]

File: app/src/main/java/com/lumisdinos/gameoffifteen/data/GameRepositoryImpl.kt (hunk 1 above)
Comment (LumisD) [reply app/src/main/java/com/lumisdinos/gameoffifteen/data/GameRepositoryImpl.kt:15]: done

**Comment by artemMprokhorov** [comment]
now - ok

File: app/src/main/java/com/lumisdinos/gameoffifteen/di/ApplicationComponent.kt [hunk 2]
```diff
@@ -12,10 +12,7 @@ import javax.inject.Singleton
 @Component(
     modules = [
         AndroidSupportInjectionModule::class,
-        MainModule::class,
-        HomeModule::class,
-        ListModule::class,
-        SettingsModule::class
+        PresentationModule::class
```
Comment (artemMprokhorov) [inline app/src/main/java/com/lumisdinos/gameoffifteen/di/ApplicationComponent.kt:15]: now is ok

**Review by artemMprokhorov — APPROVED** [review]

**Pull request merged**